### PR TITLE
fix tpo registration tree crash issue when select plant project from …

### PR DESCRIPTION
**Pull request by sanjayradadiya**
…dropdown

Fixes #

Changes in this pull request:
- Fix tpo tree registration plant project selection issue
- Fix undefined api call when click on specific tree card

@norbertschuler @harshvitra

### CODE REVIEW

**Comment by sanjayradadiya** [comment]
https://www.loom.com/share/88cd83109a164a8d86aaad6e579b8917

**Comment by norbertschuler** [comment]
- [ ] I tried to register a tree with the second project in the dropdown, but as far as I could see from the API https://app-development.plant-for-the-planet.org/api/v1.1/de/loadUserProfile the user contribution was associated with the first project:

Bildschirmfoto 2020-04-23 um 18 57 44

File: app/components/RegisterTrees/formComponents.native.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ export const FormikFormTree = props => {
         : '';
     }
   };
-
+console.log('validationSchema',validationSchema)
```
Comment (norbertschuler) [inline app/components/RegisterTrees/formComponents.native.js:100]: please remove before merging

File: app/components/RegisterTrees/formComponents.native.js [hunk 2]
```diff
@@ -159,6 +159,7 @@ export const FormikFormTree = props => {
                               </Text>
                             </TouchableWithoutFeedback>
                           </Text>
+                          {console.log('formik props',props)}
```
Comment (norbertschuler) [inline app/components/RegisterTrees/formComponents.native.js:162]: please remove before merging

**Comment by norbertschuler** [comment]
- [ ] I found one more issue not directly related, but also a bug: If you choose the second project of a TPO, e.g. `203` in my test data and then edit your user contribution (registered tree) once more, the edit form does not contain the project dropdown any more and if you save it, the project id is reset to the first project of the TPO, e.g. `201` in my test data. I am fine with not allowing to change the project for the user contribution any more, but we should not automatically change the project id when editing.

**Comment by norbertschuler** [comment]
> * [ ]  I found one more issue not directly related, but also a bug: If you choose the second project of a TPO, e.g. `203` in my test data and then edit your user contribution (registered tree) once more, the edit form does not contain the project dropdown any more and if you save it, the project id is reset to the first project of the TPO, e.g. `201` in my test data. I checked that the wrong project id is send as payload to the API call PUT https://app-development.plant-for-the-planet.org/api/v1.3/de/plantContributions/307220 to store the changed data:

This is probably related to https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/2236#issuecomment-619743578 not given back a plantProject id for the result of registering a tree.

**Comment by sanjayradadiya** [comment]
https://www.loom.com/share/77593bbe8e364a5c8fb0eca09b0acf77

**Pull request merged**